cf-agent/cf-agent.c

```diff
@@ -308,6 +310,11 @@ static GenericAgentConfig *CheckOpts(int argc, char **argv)
             TIMING = true;
             break;
 
+        case 'w':
+            Log(LOG_LEVEL_INFO, "Setting workdir to '%s'", optarg);
+            putenv("CFENGINE_TEST_OVERRIDE_WORKDIR", xstrdup(optarg), 1);
```

Comment:
Oh, sorry. I didn't push the latest change. I've done so now, amended to use `string_lib.h` (a local function in `cf-agent.c` had to be renamed).